todo-app/src/todo/controller.ts

```diff
@@ -0,0 +1,54 @@
+import {
+    Body,
+    Controller,
+    Delete,
+    Get,
+    Param,
+    ParseIntPipe,
+    Post,
+    Put,
+  } from '@nestjs/common';
+  import { TodoService } from './service';
+  import {
+    CreateTodoRequestDto,
+    GetTodoDateRequestDto,
+  } from './dto/request.dtos';
+  import { TodoResponseDto } from './dto/response.dtos';
+  
+  @Controller('todo')
+  export class TodoController {
+    constructor(private todoService: TodoService) {}
+  
+    // create todo
+    @Post()
+    createTodo(
+      @Body() requestDto: CreateTodoRequestDto,
+    ): Promise<TodoResponseDto> {
+      return this.todoService.createTodo(requestDto);
+    }
+  
+    // get all todos for the username̦
+  
+    @Get('/:userName')
+    getTodoByUsername(
+      @Param('userName') userName: string,
+    ): Promise<TodoResponseDto[]> {
+      return this.todoService.getTodoByUsername(userName);
+    }
+  
+    // delete todo for a username
+    @Delete('/:id')
+    deleteTodoByID(@Param('id', new ParseIntPipe()) id: number): Promise<void> {
```

Comment:
- returning void makes it difficult for UI to understand what happened
- always try to return a simple success response in these cases